source/simulator/lateral_averaging.cc

```diff
@@ -120,7 +120,19 @@ namespace aspect
     Utilities::MPI::sum(values, this->get_mpi_communicator(), values_all);
 
     for (unsigned int i=0; i<num_slices; ++i)
-      values[i] = values_all[i] / (static_cast<double>(volume_all[i])+1e-20);
+      {
+        if (volume_all[i] > 0.0)
+          {
+            values[i] = values_all[i] / (static_cast<double>(volume_all[i])+1e-20);
+          }
+        else
+          {
+            // Output nan if no quadrature points in depth block
+            this->get_pcout() << "\tWarning: No quadrature points found in depth block " << i
```

Comment:
The question is: "Once we get here, can the code ever continue to produce anything sensible?" If the answer is no, then it should be an `AssertThrow`. If yes, but some of the data is invalid, then we could discuss whether a warning is appropriate, or whether we should also just abort the program. What would a user do with this data?

I'm generally not overly fond of warnings. If we think the input is bad and consequently we can't compute what is requested, then my instinct is to just punt and abort, rather than figure out what subset of the problem we can still solve.